### PR TITLE
[SCB-1335] Upgrade docker-compose file to 3.0

### PR DESCRIPTION
**Pull request by chanjarster**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

Upgrade demo's docker-compose files from version 2.1 to 3.0.

Please notice that conditional `depends_on` which is add in 2.1 [\[1\]][1] is removed from 3.0 [\[2\]][2]. So I change the following like below:

```yaml
depends_on:
  database:
    condition: service_healthy
```

to like this:

```yaml
depends_on:
  - database
```

I tested the modified docker-compose file on my machine, that change also works except that there are some retry log like this:

```txt
2019-06-30 02:45:56.059 ERROR 1 --- [ault-executor-2] o.a.s.p.o.c.g.c.ReconnectStreamObserver  : Failed to process grpc coordinate command.
stacks...
2019-06-30 02:45:56.291  INFO 1 --- [pool-2-thread-1] .a.s.p.o.c.g.c.PushBackReconnectRunnable : Retry connecting to alpha at alpha-server.servicecomb.io:8080
2019-06-30 02:45:56.333 ERROR 1 --- [pool-2-thread-1] .a.s.p.o.c.g.c.PushBackReconnectRunnable : Failed to reconnect to alpha at alpha-server.servicecomb.io:8080
stacks...
2019-06-30 02:46:02.359  INFO 1 --- [pool-2-thread-1] .a.s.p.o.c.g.c.PushBackReconnectRunnable : Retry connecting to alpha at alpha-server.servicecomb.io:8080
2019-06-30 02:46:02.527  INFO 1 --- [pool-2-thread-1] .a.s.p.o.c.g.c.PushBackReconnectRunnable : Retry connecting to alpha at alpha-server.servicecomb.io:8080 is successful

```

[1]: https://docs.docker.com/compose/compose-file/compose-file-v2/#depends_on
[2]: https://docs.docker.com/compose/compose-file/#depends_on

### CODE REVIEW

**Comment by WillemJiang** [comment]
@chanjarster 
I just checked the document, it looks like health check doesn't work any more,  we need to use [sh-compatible wait-for](https://docs.docker.com/compose/startup-order/) to verify the service availability before starting the docker image.

I think in this way we could avoid the errors you hit.  
BTW, we could remove the health-check part, once we upgrade the docker compose file to 3.x.

**Comment by chanjarster** [comment]
Thanks for the suggestion. If using wait-for in docker-compose file like this:

```yaml
command: ["./wait-for", "alpha:8080", "--", "java", "-jar", "..."]
```

The `"java", "-jar", "..."` part will be a duplicate of `ENTRYPOINT` line in Dockerfile. So maybe I should write a `docker-entrypoint.sh`?

**Comment by WillemJiang** [comment]
> Thanks for the suggestion. If using wait-for in docker-compose file like this:
> 
> ```yaml
> command: ["./wait-for", "alpha:8080", "--", "java", "-jar", "..."]
> ```
> 
> The `"java", "-jar", "..."` part will be a duplicate of `ENTRYPOINT` line in Dockerfile. So maybe I should write a `docker-entrypoint.sh`?

Yeah, I agree, we could add a shell to wrap the wait-for command, and we need to make sure those shell are placed in the PATH.

**Comment by chanjarster** [comment]
@WillemJiang Please review the new commit.

**Comment by WillemJiang** [comment]
It looks good to me. We just need to wait for the CI build.

**Comment by WillemJiang** [comment]
CI build are passed, @coolbeevip @zhfeng do you have any comments on this PR?
@chanjarster 's last commit put the wait-for as the start entry point of docker images, so we can start the service after the depended service is started.

File: docker-build-config/docker-entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+
```
Comment (WillemJiang) [inline docker-build-config/docker-entrypoint.sh]: We to add the Apache license header for this file.

**Comment by chanjarster** [comment]
@WillemJiang License header added

**Pull request merged**